toggl/utils/others.py

```diff
@@ -175,3 +179,8 @@ def toggl(url, method, data=None, headers=None, config=None, address=None):
 
     # If retries failed then 'e' contains the last Exception/Error, lets re-raise it!
     raise exception
+
+
+if Config.factory().cache_requests:
```

Comment:
Hmm I am afraid we can't do this. `Config.factory()` will always give you the default `~/.togglrc` configuration, which might not be desired for people who build their own `Config` instance that they pass to the API Wrappers.

Reply:
Hmm we could wrap instead of `toggl()` the `_toggl_request()` function and then in `toggl()` decide based on the passed `config` if used the cached version or instead non-cached version using the ability to bypass the cache using `_toggl_request. __wrapped__()` attribute that `lru_cache` expose. This then can be also used to ignore cache for POST/DELETE/PUT requests.

But still, it leaves us with the problem of caching error response...